manifests/repo.pp

```diff
@@ -1,32 +1,53 @@
-# Configure package repository
+# @api private
+# @summary Configure package repository
 #
-class php::repo {
-  $msg_no_repo = "No repo available for ${facts['os']['family']}/${facts['os']['name']}"
+# @param [Boolean] manage_external_repo
+#   manage an external package repository or not
+# @option external_repo [Boolean] :false
+#   default
+# @option external_repo [Boolean] :true  
+#
+# @param [Array] external_repo_supported_php_versions
+#   The PHP versions the additional repositories support
+# @option external_repo_supported_php_versions [Array] :[]
+#   default: empty array, default for OS from Hiera Data in module
+#
+# @param [Boolean] manage_internal_repo
+#   manage an internal package repository, parameter is to control
+#   app streams in CENTOS/RHEL/Fedora, Repos on FreeBSD or Archlinux
+# @option external_repo [Boolean] :false
+#   true
+# @option external_repo [Boolean] :true   
+#
+# @param [Array] os_supported_php_versions
+#   The PHP versions the OS supports by default, without any additional repositories,
+#   on RHEL/CentOS and Fedora this includes app streams, on FreeBSD or Archlinux it manages Repos
+# @option os_supported_php_versions [Array] :[]
+#   default: empty array, default for OS from Hiera Data in module
+# 
+# @param [String] php_version
+#   Set the PHP version to install
+# @option php_version [String] :'7.4'
+#   defaults: latest stable PHP Version available on php.net, default for OS from Hiera Data in module
+#
+class php::repo (
+  Boolean $manage_internal_repo                         = $php::manage_internal_repo,
```

Comment:
In my opinion yes, because there are operating systems where we need to manage internal and external repos (RHEL/CENTOS/Fedora)  app channel/Remi and if I want to have different data without interpolation this seems to be the way to go

Reply:
don't know about centos & co, you manage the internal and the external repo for php?

is it relevant if it is external or internal? isn't it just multiple `repositories`?

Reply:
Its been a while since I worked with Fedora based systems, it only affects RHEL8/CENTOS8 and Fedora IDK which version RHEL8 was branched from +. For me this seems important, because it will prevent users declaring internal and external php versions/sources parallel.